raiden/network/proxies/payment_channel.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+from typing import Dict
+
+from raiden.network.proxies import TokenNetwork
+from raiden.utils import typing
+
+
+class PaymentChannel:
+    def __init__(
+        self,
+        token_network: TokenNetwork,
+        partner_address: typing.Address,
+    ):
+        self.token_network = token_network
+        self.partner_address = partner_address
+
+    def token_address(self) -> typing.Address:
+        """ Returns the address of the token for the channel. """
+        return self.token_network.token_address()
+
+    def channel_identifier(self) -> typing.ChannelID:
+        """ Returns the channel identifier. """
+        return self.token_network.detail_channel(self.partner_address)['channel_identifier']
+
+    def detail(self) -> Dict:
+        """ Returns the channel details. """
+        return self.token_network.detail(self.partner_address)
+
+    def settle_block_number(self) -> typing.BlockNumber:
+        """ Returns the settle block of the channel"""
+        return self.token_network.settle_block_number(self.partner_address)
```

Comment:
s/settle_block_number/settle_timeout/

these are not equivalent, as the token network settle block number is either the timeout before the channel is closed or the expiration once close is called.

Reply:
Would it be ok to have both `settle_block_number` and `settle_timeout` and:
- if the channel state is `open`: return the on-chain value for `settle_timeout` and `None` for `settle_block_number`
- if the channel state is `closed`: return `None` for `settle_timeout` and the on-chain value for `settle_block_number`

Reply:
I would say no.

The goal for this is really to save us some work, we just need to make a compatible api for now and remove this later, returning `None` if the channel is closed is not a compatible api, not sure if this is needed in the code base

Reply:
Ok, you want an exact replacement (as the title actually says). So: keep `settle_timeout` and not `settle_block_number`

Reply:
@loredanacirstea Is there a way to get the `settle_timeout` value for a channel from the contracts? Or do we have to remember this from the `ChannelOpened` event?

Reply:
@palango , remembering that from the `ChannelOpened` event would be the sure thing. Otherwise, calling `settle_block_number` will return the initial `settle_timeout` value when the channel is open. After the `closeChannel` transaction is sent, the contract will add the block number to the timeout, so `settle_block_number` will return the block number after which `settleChannel` can be called.

Reply:
> Otherwise, calling settle_block_number will return the initial settle_timeout value when the channel is open. After the closeChannel transaction is sent, the contract will add the block number to the timeout, so settle_block_number will return the block number after which settleChannel can be called.

I think we want to avoid this especially.